lib/components/FileAttachment/index.jsx

```diff
@@ -0,0 +1,181 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import "@uppy/core/dist/style.css";
+import "@uppy/progress-bar/dist/style.css";
+import { saveAs } from "file-saver";
+import { isEmpty } from "ramda";
+
+import directUploadsApi from "apis/direct_uploads";
+import { DIRECT_UPLOAD_ENDPOINT } from "common/constants";
+import Button from "components/Common/Button";
+import useUppyUploader from "hooks/useUppyUploader";
+
+import Attachments from "./Attachments";
+import {
+  DEFAULT_UPPY_CONFIG,
+  DROP_DOWN_OPTIONS_KEYS,
+  UPPY_UPLOAD_CONFIG,
+} from "./constant";
+
+import Progress from "../Editor/MediaUploader/Progress";
+
+const FileAttachment = ({
+  endpoint = DIRECT_UPLOAD_ENDPOINT,
+  onRename = _ => {},
+  onDelete = _ => {},
+  onError = _ => {},
+}) => {
+  const fileInputRef = useRef(null);
+  const [uploadedFiles, setUploadedFiles] = useState([]);
+  const [error, setError] = useState("");
+  const uppyConfig = { ...DEFAULT_UPPY_CONFIG };
+  const onClickAttachment = () => fileInputRef?.current?.click();
+
+  const {
+    uppy,
+    isUploading,
+    error: uppyError,
+  } = useUppyUploader({
+    endpoint,
+    uppyConfig,
+    uppyUploadConfig: { ...UPPY_UPLOAD_CONFIG },
+  });
+
+  useEffect(() => {
+    if (!isEmpty(uppyError)) {
+      setError(uppyError);
+    }
+  }, [uppyError]);
+
+  const onSelectAttachment = async e => {
+    const files = Array.from(e?.target.files);
+    try {
+      for (const file of files) {
+        uppy.addFile({
+          name: file.name,
+          type: file.type,
+          data: file,
+        });
+      }
+
+      const { successful = [], failed = [] } = await uppy.upload();
+
+      if (!isEmpty(successful)) {
+        setUploadedFiles([...uploadedFiles, ...successful]);
+      }
+
+      if (failed.length > 0) {
+        const errorMessages = failed.map(file => file?.error.message).join(",");
+        setError(errorMessages);
+      }
+    } catch (e) {
+      onError && onError(e);
+    }
+  };
+
+  const handleDelete = async file => {
+    const { signed_id: signedId } = file.response;
+
+    try {
+      await directUploadsApi.destroy(`${endpoint}/${signedId}`);
+      setUploadedFiles(
+        uploadedFiles.filter(
+          uploadedFile => uploadedFile.response.signed_id !== signedId
+        )
+      );
+      onDelete(file);
+    } catch (e) {
+      setError(`Something went wrong while deleting file: ${file.name}`);
+      onError && onError(e);
+    }
+  };
+
+  const handleRename = async (attachment, callback) => {
+    try {
+      const { fileName, signedId } = attachment;
+      const payload = {
+        blob: { filename: fileName },
+      };
+
+      const response = await directUploadsApi.create(
+        `${endpoint}/${signedId}/`,
+        payload
+      );
```

Comment:
Please create a new API for `update` action.
Refer: https://github.com/bigbinary/neeto_commons/blob/main/app/controllers/api/direct_uploads_controller.rb#L14-L16